### DISCUSSION TITLE
How to Implement Biometrics for Mobile Apps with Kratos?

### DISCUSSION THREAD

**Discussion by jaocamp** [ory/kratos#1980] [Q&A] [answered]
Hi, I have questions regarding the implementation of Biometrics. As far as I understand there are two ways to authenticate the user, via `password` or `oidc`. Is there any way to authenticate the user without informing a password, only validating a pre-registered hash referring to biometrics?

I thought about using Hooks but I couldn't generate a session token without the user informing the password.
Has anyone been through this and has any idea how to solve so I can use Kratos in my solution?

**Comment by aeneasr** [comment, marked as answer]
We don’t have such a feature yet, if you have ideas for implementation please open a design doc :)

**Comment by vinckr** [comment]
Hello @rafaelcam 
It has been a while but Ory supports passwordless [WebAuthn](https://www.ory.sh/docs/kratos/passwordless/webauthn) / [Passkeys](https://www.ory.sh/docs/kratos/passwordless/passkeys), you can use those with biometrics like fingerprint or FaceId.
